### PR TITLE
Static start_date and default arg cleanup for Microsoft providers example DAGs

### PR DESCRIPTION
**Pull request by josh-fell**
There is an ongoing effort to enhance example DAGs by setting static values for `start_date` (already complete for core example DAGs), cleaning up and/or implementing useful `default_args`, and removing/limiting the use of redundant, default connection ID values.

This PR mainly addresses these 3 cleanup areas in example DAGs across Microsoft providers.  There are other small docs updates.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/microsoft/azure/example_dags/example_azure_blob_to_gcs.py [hunk 1]
```diff
@@ -16,44 +16,33 @@
...
+# [START how_to_azure_blob_to_gcs]
 with DAG(
     "example_azure_blob_to_gcs",
     schedule_interval=None,
-    start_date=days_ago(1),  # Override to match your needs
+    start_date=datetime(2021, 1, 1),  # Override to match your needs
+    default_args={"container_name": AZURE_CONTAINER_NAME, "blob_name": BLOB_NAME},
 ) as dag:
 
-    # [START how_to_wait_for_blob]
```
Comment (josh-fell) [inline airflow/providers/microsoft/azure/example_dags/example_azure_blob_to_gcs.py:38]: Removed this tag as it was not being used in any documentation.

File: airflow/providers/microsoft/azure/example_dags/example_azure_container_instances.py [hunk 2]
```diff
@@ -27,17 +27,10 @@
...
-        'depends_on_past': False,
-        'email': ['airflow@example.com'],
-        'email_on_failure': False,
-        'email_on_retry': False,
-        'retries': 1,
-        'retry_delay': timedelta(minutes=5),
-    },
-    schedule_interval=timedelta(1),
+    default_args={'retries': 1},
+    schedule_interval=timedelta(days=1),
```
Comment (josh-fell) [inline airflow/providers/microsoft/azure/example_dags/example_azure_container_instances.py:31]: Updated `schedule_interval` to use keyword args to be more explicit.

File: airflow/providers/microsoft/winrm/example_dags/example_winrm.py [hunk 3]
```diff
@@ -28,23 +28,22 @@
...
     dag_id='POC_winrm_parallel',
     schedule_interval='0 0 * * *',
-    start_date=days_ago(2),
+    start_date=datetime(2021, 1, 1),
     dagrun_timeout=timedelta(minutes=60),
     tags=['example'],
+    catchup=False,
 ) as dag:
 
-    cmd = 'ls -l'
```
Comment (josh-fell) [inline airflow/providers/microsoft/winrm/example_dags/example_winrm.py:47]: Removed unused variable in DAG.

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_blob_to_gcs.rst [hunk 4]
```diff
@@ -51,10 +51,10 @@ Operator transfers data from Azure Blob Storage to specified bucket in Google Cl
 
 To get information about jobs within a Azure Blob Storage use:
 :class:`~airflow.providers.microsoft.azure.transfers.azure_blob_to_gcs.AzureBlobStorageToGCSOperator`
+
```
Comment (josh-fell) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_blob_to_gcs.rst:54]: Space added for better formatting.

**Review by ephraimbuddy — APPROVED** [review]
LGTM

Some of these examples are used in system tests. Can you verify them by executing the system tests in https://github.com/apache/airflow/tree/main/tests/providers/microsoft/azure/transfers.

**Pull request merged**